src/App/Document.cpp

```diff
@@ -4189,13 +4189,20 @@ const std::vector<DocumentObject*>& Document::getObjects() const
     return d->objectArray;
 }
 
-
 std::vector<DocumentObject*> Document::getObjectsOfType(const Base::Type& typeId) const
+{
+    return getObjectsOfType({typeId});
+}
+
+std::vector<DocumentObject*> Document::getObjectsOfType(const std::vector<Base::Type>& types) const

```

Comment:
At parameter call, I see that you extract the typeid first. Why not pass the types (or objects), and then call `isDerivedFrom<obejct>()` inside this function? Or get type id inside this function?

Reply:
I'm not sure to understand what you mean. I used the pre existing syntax of getObjectsOfType which is already widely used so changing this would be difficult to change now.

Reply:
We don't need to change existing functions. We just need to care for new ones. We are already migrating from isDerivedFrom(type::typeid) to isDerivedFrom<type>()

Reply:
That would be harder to implement I think as it would need variadic templates, something that is not such easy to implement.

Reply:
Let's keep it simple then